spec/models/contact_spec.rb

```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+RSpec.describe Contact, type: :model do
+  describe "associations" do
+    it { should belong_to(:signature) }
+  end
+
+  describe "validations" do
+    it { should validate_presence_of(:signature) }
+    it { should validate_presence_of(:address) }
+    it { should validate_presence_of(:phone_number) }
+    it { should validate_length_of(:address).is_at_most(500) }
+    it { should_not allow_value('6'*32).for(:phone_number).with_message(:too_long) }
+  end
+
+  describe "instance methods" do
+    describe "#phone_number=" do
+      it "normalizes input" do
+        contact = Contact.new(phone_number: "0300 200 6565")
+        expect(contact.phone_number).to eq("03002006565")
+      end
```

Comment:
That's implicitly relying on Active Record calling `#phone_number=` - better to use the method explicitly:

``` ruby
describe "#phone_number=" do
  it "normalizes input" do
    expect {
      subject.phone_number = "0300 200 6565"
    }.to change {
      subject.phone_number
    }.from(nil).to("03002006565")
  end
end
```

Reply:
I it test the setter explicitly, but tweaked the style to what I believe to be more readable. Tell me what you think and I'll change it if you want.

Reply:
Not a fan - can you change to how I posted it (which matches similar specs elsewhere).

Reply:
fair enough :sparkles:  :sparkles:  :sparkles: